src/gc.rs

```diff
@@ -337,10 +337,13 @@ impl Collector {
     fn check_pointer(&mut self, word: Word) {
         // Since the heap starts at the end of the data segment, we can use this
         // as the lower heap bound.
-        if word >= self.data_segment_end && word < unsafe { crate::alloc::HEAP_TOP } {
+        if word >= self.data_segment_end && word <= unsafe { crate::alloc::HEAP_TOP } {
             if let Some(block) = ALLOCATOR
                 .iter()
-                .find(|x| x.ptr == word || word > x.ptr && word < x.ptr + x.size)
+                // It's legal to hold a pointer 1 byte past the end of a block,
+                // but we can still use find because this will never over-run
+                // the size of the next block's header.
+                .find(|x| x.ptr == word || word > x.ptr && word <= x.ptr + x.size + 1 )
```

Comment:
I think the `+ 1` at the end is wrong: chaging `<` to `<=` is sufficient. The `+1` then allows a pointer to *2 bytes* after the end, which I don't think we should support.

Reply:
Bah! You're right. I was only testing you...

Reply:
Fixed: a6dd3254eab3480dab7506851d0d199e70077640